app/src/main/kotlin/compiler/intermediate/ControlFlowPlanner.kt

```diff
@@ -517,12 +507,12 @@ class ControlFlowPlanner(private val diagnostics: Diagnostics) {
                 }
             }
         } else {
-            expressionTypes[Ref(expression)]?.let {
-                if (it is Type.Array) {
-                    cfgBuilder.addNextCFG(arrayMemoryManagement.genRefCountDecrement(result, it))
-                }
-            }
```

Comment:
Do I understand correctly that now if we provide `accessNodeConsumer` then calling function is responsible for adding the refcount decrement?
(I guess that is not going to happen anyway)

Reply:
Do we want to decrease the reference counter if `accessNodeConsumer` is taking `it`? Doesn't that make the `accessNodeConsumer` a new owner of the current reference to `it`?

Reply:
Yes, I think that `genRefCountDecrement` should be called if and only if `accessNodeConsumer` is null, otherwise `accessNodeConsumer` becomes new owner of that reference. I thought this case isn't possible anyway, but I forgot about `Statement.GeneratorYield`. In that case, I believe we don't need to worry about refcount - we can simply keep it incremented to the point when function that called generator receives it (we don't need to increment it there).